python/ray/_raylet.pyx

```diff
@@ -184,6 +184,96 @@ def compute_task_id(ObjectID object_id):
     return TaskID(object_id.native().TaskId().Binary())
 
 
+def setup_logging(stdout_name, stderr_name):
+    """Sets up logging for the current worker, creating the file and flushing
+    buffers as is necessary.
+
+    Args:
+        stdout_name (str): The file name that stdout should be written to.
+        stderr_name(str): The file name that stderr should be written to.
+
+    Returns:
+        (tuple) The absolute paths of the files that stdout and stderr will be
+    written to.
+    """
+    stdout_path = ""
+    stderr_path = ""
+    worker_pid = os.getpid()
+
+    cdef extern from "ray/util/util.h":
+        void flush_out()
+        void flush_err()
+
+    if stdout_name:
+        # Line-buffer the output (mode 1).
+        stdout_file = open_worker_log(stdout_name, worker_pid)
+
+        # Before redirecting stdout we need to make sure userspace buffers
+        # are all flushed.
+        # Flush python's userspace buffers
+        sys.stdout.flush()
+
+        # TODO (Alex): Flush the c/c++ userspasce buffers if necessary.
+        # `fflush(stdout); cout.flush();`
+
+        stdout_fileno = sys.stdout.fileno()
+
+        # Redirect stdout at the file descriptor level. If we simply set
+        # sys.stdout, then logging from C++ can fail to be
+        # redirected. Note that dup2 will automatically close the old file
+        # descriptor before overriding it.
+        os.dup2(stdout_file.fileno(), stdout_fileno)
```

Comment:
This one can't error because we're the ones opening the file.

Reply:
Oh I see. Maybe put this as a postcondition of `open_worker_log` (doc + `assert`)? Right now it just says the return value is a "file-like object" which seems to imply it doesn't necessarily need to be bound to a file descriptor.

Reply:
I say "file-like object" because I think in the long term we should actually replace this with named pipes.

Reply:
Right, that make sense. Just something to indicate it should have a `fileno()` would be nice since right now you can't tell just by looking at the function; it's an implicit requirement.